src/common/SIRF.py

```diff
@@ -98,7 +99,10 @@ def dot(self, other):
         data viewed as vectors.
         other: DataContainer
         '''
-        assert_validities(self, other)
+        assert_validities(self,other)
+        # Check if input are the same size
+        if numpy.prod(self.dimensions()) == numpy.prod(other.dimensions()):
```

Comment:
surely you want `!=` here

Reply:
Indeed! my mistake, was testing, left the wrong version.